src/main/java/frc/robot/subsystems/Climber.java

```diff
@@ -14,131 +14,79 @@
-  public CommandBase moveCurrent(DoubleSupplier speed) {
-    return cmd("Move With Current Limit").onExecute(() -> {
-      motor.set(validatorLimit.applyAsDouble(speed.getAsDouble()));
-      SmartDashboard.putNumber("Climber Speed", speed.getAsDouble());
-    }).finishedWhen(motor::errored).onEnd((interrupted) -> {
-      motor.set(0.0);
-    });
-  }
+    private final double direction;
 
-  // Move the motor with variable speed and stop when limit switches are hit
-  public CommandBase moveLimit(DoubleSupplier speed) {
-    return cmd("Move With Limits").onExecute(() -> {
-      motor.set(switchLimit.run(speed.getAsDouble()));
-      SmartDashboard.putNumber("Climber Speed", speed.getAsDouble());
-    }).onEnd((interrupted) -> {
-      motor.set(0.0);
-    });
-  }
-
-  public CommandBase extend() {
-    return cmd("Extend").onExecute(() -> {
-      motor.set(limit.run(EXTEND_SPEED));
-    }).onEnd((interrupted) -> {
-      motor.set(0.0);
-    });
-  }
+    private ExtendDirection(double direction) {
+      this.direction = direction;
+    }
 
-  // Extend the climber and use validators stop the motors
-  public CommandBase extendCurrent() {
-    return cmd("Extend With Current Limit").onExecute(() -> {
-      motor.set(validatorLimit.applyAsDouble(EXTEND_SPEED));
-    }).finishedWhen(motor::errored).onEnd((interrupted) -> {
-      motor.set(0.0);
-    });
+    public double get() {
+      return direction;
+    }
   }
 
-  // Extend the motor with limit switches
-  public CommandBase extendLimit() {
-    return cmd("Extend With Limits").onExecute(() -> {
-      motor.set(switchLimit.run(EXTEND_SPEED));
-    }).onEnd((interrupted) -> {
-      motor.set(0.0);
-    });
-  }
+  public Climber(ClimberMap map) {
 
-  public CommandBase retract() {
-    return cmd("Retract").onExecute(() -> {
-      motor.set(limit.run(RETRACT_SPEED));
-    }).onEnd((interrupted) -> {
-      motor.set(0.0);
-    });
-  }
+    extendMotor = map.getExtendMotor();
+    rotateMotor = map.getRotateMotor();
 
-  public CommandBase retractCurrent() {
-    return cmd("Retract With Current Limit").onExecute(() -> {
-      motor.set(validatorLimit.applyAsDouble(RETRACT_SPEED));
-    }).finishedWhen(motor::errored).onEnd((interrupted) -> {
-      motor.set(0.0);
-    });
+    extendLimit = Modifier.unless(extendMotor::errored);
+    rotateLimit = Modifier.unless(extendMotor::errored);
   }
 
-  // Retract the motor with limit switches
-  public CommandBase retractLimit() {
-    return cmd("Retract With Limits").onExecute(() -> {
-      motor.set(switchLimit.run(RETRACT_SPEED));
-    }).onEnd((interrupted) -> {
-      motor.set(0.0);
+  // Move the motor based off a variable speed
+  public CommandBase extendSpeed(DoubleSupplier speed) {
+    return cmd("Extend Speed").onExecute(() -> {
+      double nSpeed = speed.getAsDouble();
+      extendMotor.set(extendLimit.applyAsDouble(nSpeed));
+      SmartDashboard.putNumber("Climber Speed", nSpeed);
```

Comment:
It would be good to also send the speed we set after the modifiers. You can call get on the motor controller to do so.